Don't use the insecure git:// protocol when fetching git gems

strugee opened the pull request:
Fixes #1756

dsander commented on Gemfile.lock:
```diff
@@ -1,14 +1,14 @@
-  remote: git://github.com/amatsuda/kaminari.git
-  revision: abbf93d557208ee1d0b612c612cd079f86ed54f4
+  remote: https://github.com/amatsuda/kaminari.git
+  revision: e83d6cebf69fd78e3c9830992209601fdc3e0f23
```
Can you either manually change the revisions to the previous one, or lock it in the `Gemfile`?

dsander replied:
Please run `bundle install` to update the `Gemfile.lock`

dsander commented on app/.bundle/config:
```diff
@@ -0,0 +1,2 @@
+---
```
Doesn't that file have to be in the repository root instead of inside `app`?

strugee replied:
Not according to `bundle config --help`.

dsander replied:
Right, I think that example configuration location is misleading with `app` they probably mean `your-applications-root-path`. I had to move the config directive into `.bundle/config` to install the gems via https.

strugee replied:
@dsander weird, it worked on my machine with the file in `app/.bundle`... ah well! Works for me this way too, so I made that change in the patchset I just pushed.

strugee commented:
@dsander not sure why... let's see if it still happens with the rebased commits I just force-pushed

strugee commented:
Apparently it still doesn't work... Bundler version too old, maybe? What version do the Docker images run?

dsander commented:
I did some testing locally, somehow I can not create a pull request for your fork. This commit should have the required changes for the Dockerfiles: https://github.com/dsander/huginn/commit/63fefac10e5f0665d8b46981b80d0f723f0146a9

strugee commented:
@dsander sorry, totally forgot about this PR (until I updated Huginn and got these warnings again :P)

I pulled your patch into the PR (modified to not touch `.bundle/config`, since I fixed that in an earlier commit), so the Docker builds should work now.

dsander commented:
@strugee No worries :) I now think we should now remove the `.bundle/config` from the git ignored files. Depending on the installation the user might have custom settings in there which get overridden when checking out the current code.
I found a nice snipped which overrides the `git_source` method to ensure `github` remotes always use HTTPS. Feel free to squash my commit into yours if you like it.

dsander commented:
Send you a PR on your fork because I could not push to that branch.

strugee commented:
@dsander apparently I forgot to ping you when I merged your PR. I've just rebased on master to fix merge conflicts - this should be ready for merge now

strugee commented:
Nevermind, I bungled this. Will ping you when it's ready

strugee commented:
@dsander just pushed the fixed branch. Please merge, assuming CI passes :)

dsander commented:
Hey, it does not look like the commits from https://github.com/strugee/huginn/pull/1/commits are in this branch.

strugee commented:
@dsander shoot, you're right. Now, I _did_ pull in the change from https://github.com/dsander/huginn/commit/63fefac10e5f0665d8b46981b80d0f723f0146a9, and that _did_ fix the Docker builds to use HTTPS. Can't remember why I didn't take your patches, though.

Since Docker is fixed anyway, is there any reason to fully merge the PR you sent? I can't see any, but if you do I'm happy to fix it up.

dsander commented:
@strugee Sorry I oversaw the notification. Yes https://github.com/dsander/huginn/commit/63fefac10e5f0665d8b46981b80d0f723f0146a9 was my first approach, but I missed that we can not modify `.bundle/config` because that file also has local bundle configuration that is different from system to system (production differs from development installations). 
In https://github.com/strugee/huginn/pull/1/files I reverted those changes and modified the Gemfile to ensure we are always using HTTPS.
You should be able to apply them to your branch by cherry-picking https://github.com/dsander/huginn/commit/e01f94ca463a57d159b44290de8c6079a641d809 and https://github.com/dsander/huginn/commit/87bcdcd7596581bd162f94dfeef2596f7d78dcee

strugee commented:
@dsander OK, fixed. In that case, should I remove the commit touching `.bundle/config`? I should think so, right?

dsander commented:
Nice, yes please remove it.

strugee commented:
@dsander done.

dsander commented on Gemfile.lock:
```diff
@@ -38,16 +38,16 @@ GIT
 
 GIT
-  remote: git://github.com/tumblr/tumblr_client.git
-  revision: 0c59b04e49f2a8c89860613b18cf4e8f978d8dc7
```
Can you lock the ref to `0c59b04e49f2a8c89860613b18cf4e8f978d8dc7` in the Gemfile?

dsander commented:
@strugee Thanks for staying on this for so long 😄 

When I run `bundle install` I am getting a diff for `Gemfile.lock`, does that not happen for you?

```
diff --git a/Gemfile.lock b/Gemfile.lock
index cfac087..30d21c2 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -1,6 +1,7 @@
 GIT
   remote: https://github.com/amatsuda/kaminari.git
   revision: abbf93d557208ee1d0b612c612cd079f86ed54f4
+  ref: abbf93d557208ee1d0b612c612cd079f86ed54f4
   branch: 0-17-stable
   specs:
     kaminari (0.17.0)
@@ -48,6 +49,7 @@ GIT
 GIT
   remote: https://github.com/tumblr/tumblr_client.git
   revision: 0c59b04e49f2a8c89860613b18cf4e8f978d8dc7
+  ref: 0c59b04e49f2a8c89860613b18cf4e8f978d8dc7
   branch: master
   specs:
     tumblr_client (0.8.5)
```

strugee commented:
@dsander that happens for me too; what's in this PR matches `master`: https://github.com/cantino/huginn/blob/master/Gemfile.lock

dsander commented:
If your `Gemfile.lock` also includes the changes I mentioned we need to have them in this PR too. Not sure why `ref` and `revision` are duplicated, but it does not seem to cause any issues.

strugee commented:
@dsander OK, fixed. Weird that `ref` is only added on two Git sources, though, and not all of them.

dsander commented:
@strugee Nice, thanks a lot!

I think the `ref` is added when the revision is explicitly set in the Gemfile.

strugee commented:
@dsander thanks for sticking with this! 🎆 🎊 💯

cantino commented:
Awesome!
